src/views/login.js

```diff
@@ -7,5 +7,7 @@ if (params.has('auth')) {
         message.innerText = 'You must log in to do that.';
     } else if (status === 'failed') {
         message.innerText = 'Wrong username or password.';
+    } else if (status === 'expired') {
```

Comment:
where does that come from?

Reply:
`src/routes/loginUtilities.js:16`. You know, in a totally separate place than either of the other two searchParams are created. IDK, seemed like a good idea to have a function that could validate and redirect on its own. Let me re-think and re-write this. Pattern suggestions are welcome.

Reply:
Okay, I think this is better! I did more reading on the Fastify request lifecycle, and put authentication in a preHandler hook, where it belongs. All of the error redirects are in `routes/loginUtilities`, aside from one that's still in `routes/login`, but it seems the best place to me since it's a totally different flow. Let me know what you think, and I can do more work to make things clearer if need be.

Reply:
> Pattern suggestions are welcome.

Yes sorry for not giving any yesterday. Was tired. Still am.
I won't reread now, I hope I can review soon.
Basically I think we need a constant somewhere that we can import. Hoping that a string does not change is dangerous.

Reply:
Not a problem! Work just got busier for me, btw. I'll be Coding Lite for a week or so.

Yeah, I might even spring for an Enum in Python. Constants are the way to go, I can do that for sure.

Reply:
c410f850d2158103076f26d4674fca0031213a7e is even better than constants.